app/src/main/java/com/aarevalo/tasky/agenda/di/TaskyAgendaModule.kt

```diff
@@ -0,0 +1,48 @@
+package com.aarevalo.tasky.agenda.di
+
+import com.aarevalo.tasky.BuildConfig
+import com.aarevalo.tasky.agenda.data.remote.api.TaskyAgendaApi
+import com.aarevalo.tasky.agenda.data.repository.AgendaRepositoryImpl
+import com.aarevalo.tasky.agenda.domain.repository.AgendaRepository
+import com.aarevalo.tasky.core.domain.preferences.SessionStorage
+import com.squareup.moshi.Moshi
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.create
+import javax.inject.Named
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object TaskyAgendaModule {
+
+    @Provides
+    @Singleton
+    fun provideAgendaApi(
+        moshi: Moshi,
+        @Named("authenticated")client: OkHttpClient
+    ): TaskyAgendaApi {
+        return Retrofit.Builder()
+            .baseUrl(BuildConfig.API_BASE_URL)
+            .addConverterFactory(MoshiConverterFactory.create(moshi))
+            .client(client)
+            .build()
+            .create()
+    }
+
+    @Provides
+    @Singleton
+    fun provideAgendaRepository(
+        api: TaskyAgendaApi,
+        sessionStorage: SessionStorage
+    ) : AgendaRepository {
+        return AgendaRepositoryImpl(api, sessionStorage)
```

Comment:
For specific implementations of an interface, I'd always be as specific as possible about what makes that implementation unique. So instead of giving it the suffix `Impl`, it could be more descriptive to name it something like `OfflineFirstAgendaRepository` or including the specific library used in it (e.g. `KtorNoteDataSource`). This contains more context than a simple `Impl`.